src/mqt/predictor/rl/actions.py

```diff
@@ -0,0 +1,445 @@
+
+    OPT = "optimization"
+    SYNTHESIS = "synthesis"
+    MAPPING = "mapping"
+    LAYOUT = "layout"
+    ROUTING = "routing"
+    FINAL_OPT = "final_optimization"
+    TERMINATE = "terminate"
+
+
+@dataclass
+class Action:
+    """Base class for all actions in the reinforcement learning environment."""
+
+    name: str
+    origin: CompilationOrigin
+    pass_type: PassType
+    transpile_pass: (
+        list[qiskit_BasePass | tket_BasePass]
+        | Callable[..., list[qiskit_BasePass | tket_BasePass]]
+        | Callable[
+            ...,
+            Callable[..., tuple[Any, ...] | Circuit],
+        ]
+    )
+
+
+@dataclass
+class DeviceIndependentAction(Action):
+    """Action that represents a static compilation pass that can be applied directly."""
+
+
+@dataclass
+class DeviceDependentAction(Action):
+    """Action that represents a device-specific compilation pass that can be applied to a specific device."""
+
+    transpile_pass: (
+        Callable[..., list[qiskit_BasePass | tket_BasePass]]
+        | Callable[
+            ...,
+            Callable[..., tuple[Any, ...] | Circuit],
+        ]
+    )
+    do_while: Callable[[dict[str, Circuit]], bool] | None = None
+
+
+# Registry of actions
+_ACTIONS: dict[str, Action] = {}
+
+
+def register_action(action: Action) -> Action:
+    """Registers a new action in the global actions registry."""
+    if action.name in _ACTIONS:
+        msg = f"Action with name {action.name} already registered."
+        raise ValueError(msg)
+    _ACTIONS[action.name] = action
+    return action
+
+
+register_action(
+    DeviceIndependentAction(
+        "Optimize1qGatesDecomposition",
+        CompilationOrigin.QISKIT,
+        PassType.OPT,
+        [Optimize1qGatesDecomposition()],
+    )
+)
+
+register_action(
+    DeviceIndependentAction(
+        "CommutativeCancellation",
+        CompilationOrigin.QISKIT,
+        PassType.OPT,
+        [CommutativeCancellation()],
+    )
+)
+
+register_action(
+    DeviceIndependentAction(
+        "CommutativeInverseCancellation",
+        CompilationOrigin.QISKIT,
+        PassType.OPT,
+        [CommutativeInverseCancellation()],
+    )
+)
+
+register_action(
+    DeviceIndependentAction(
+        "RemoveDiagonalGatesBeforeMeasure",
+        CompilationOrigin.QISKIT,
+        PassType.OPT,
+        [RemoveDiagonalGatesBeforeMeasure()],
+    )
+)
+
+register_action(
+    DeviceIndependentAction(
+        "InverseCancellation",
+        CompilationOrigin.QISKIT,
+        PassType.OPT,
+        [InverseCancellation([XGate(), ZGate()])],
+    )
+)
+
+register_action(
+    DeviceIndependentAction(
+        "OptimizeCliffords",
+        CompilationOrigin.QISKIT,
+        PassType.OPT,
+        [OptimizeCliffords()],
+    )
+)
+
+register_action(
+    DeviceIndependentAction(
+        "Opt2qBlocks",
+        CompilationOrigin.QISKIT,
+        PassType.OPT,
+        [Collect2qBlocks(), ConsolidateBlocks()],
+    )
+)
```

Comment:
I am not 100% sure but I believe modern versions of Qiskit have a newer version of this block optimization that might be worth using.

Reply:
I added the `UnitarySynthesis` as it is usually done by Qiskit.